improver/utilities/cube_checker.py

```diff
@@ -203,3 +203,41 @@ def spatial_coords_match(cubes: Union[List, CubeList]) -> bool:
             and match
         )
     return match
+
+
+def assert_time_coords_valid(inputs: List[Cube], time_bounds: bool):
+    """
+    Raises appropriate ValueError if
+
+    - Any input cube has or is missing time bounds (depending on time_bounds)
+    - Input cube times and forecast_reference_times do not match
+
+    Args:
+        inputs:
+            List of Cubes where times should match
+        time_bounds:
+            When True, time-bounds are checked for and compared on the input cubes.
+            When False, the absence of time-bounds is checked for.
+
+    Raises:
+        ValueError: If any of the stated checks fail.
+    """
+    if len(inputs) <= 1:
+        raise ValueError(f"Need at least 2 cubes to check. Found {len(inputs)}")
+    cubes_not_matching_time_bounds = [
+        c.name() for c in inputs if c.coord("time").has_bounds() != time_bounds
+    ]
+    if cubes_not_matching_time_bounds:
+        str_bool = "" if time_bounds else "not "
+        msg = f"{' and '.join(cubes_not_matching_time_bounds)} must {str_bool}have time bounds"
+        raise ValueError(msg)
+    for time_coord_name in ["time", "forecast_reference_time"]:
```

Comment:
Possibly! I wasn't sure, but I've decided that it really ought to be, so the logic now ignores blend_time if it finds it on the first cube, and checks forecast_reference_time otherwise. I haven't checked for the case where one cube has both as I think we shouldn't reject those. I chose to not match blend_time values as I think it likely that we will want to combine blended data that were generated at slightly different times but are valid for the same time.